src/api/ammPools.ts

```diff
@@ -0,0 +1,31 @@
+import { PoolId } from '@ergolabs/ergo-dex-sdk';
+import { map, Observable, publishReplay, refCount, switchMap } from 'rxjs';
+
+import { AmmPool } from '../common/models/AmmPool';
+import { selectedNetwork$ } from '../network/network';
+
+export const ammPools$ = selectedNetwork$.pipe(
+  switchMap((network) => network.ammPools$),
+  publishReplay(1),
+  refCount(),
+);
+
+export const getAmmPoolById = (
+  ammPoolId: PoolId,
+): Observable<AmmPool | undefined> =>
+  ammPools$.pipe(
+    map((pools) => pools.find((position) => position.id === ammPoolId)),
+  );
+
+const byPair = (xId: string, yId: string) => (p: AmmPool) =>
```

Comment:
Bag naming for the function. It must describe an action.

Reply:
this functions works with array.filter fn. So we get something like filter(byPair).